### PR TITLE
Stream large bodies warn with modify body

### PR DESCRIPTION
**Pull request by rosydawn6**
#### Description

Added alert logging for stream_large_bodies and modify_body and enhance documentation around these options ( #6479 )

#### Checklist

 - [NA] I have updated tests where applicable.
 - [Y] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: mitmproxy/addons/modifybody.py [hunk 1]
```diff
@@ -20,7 +21,7 @@ def load(self, loader):
             """
             Replacement pattern of the form "[/flow-filter]/regex/[@]replacement", where
             the separator can be any character. The @ allows to provide a file path that
-            is used to read the replacement string.
+            is used to read the replacement string.  Streamed bodies cannot be modified.
```
Comment (mhils) [inline mitmproxy/addons/modifybody.py:24]: I think this addition here may add a bit confusion as it's not clear that this refers to mitmproxy's off-by-default streaming functionality and not say any chunked HTTP encoding. I'd be fine with leaving this out here entirely and just document it in the `stream_large_bodies` option.

File: mitmproxy/addons/modifybody.py [hunk 2]
```diff
@@ -30,6 +31,7 @@ def configure(self, updated):
             for option in ctx.options.modify_body:
                 try:
                     spec = parse_modify_spec(option, True)
+                    logging.log(ALERT, "Cannot modify streamed bodies.")
```
Comment (mhils) [inline mitmproxy/addons/modifybody.py:34]: As you said, this is just duplicating - let's take it out.

File: mitmproxy/addons/proxyserver.py [hunk 3]
```diff
@@ -211,6 +213,7 @@ def configure(self, updated) -> None:
         if "stream_large_bodies" in updated:
             try:
                 human.parse_size(ctx.options.stream_large_bodies)
+                logging.log(ALERT, "Cannot modify streamed bodies.")
```
Comment (mhils) [inline mitmproxy/addons/proxyserver.py:216]:
```suggestion
                logging.log(ALERT, "Both modify_body and stream_large_bodies are active. Streamed bodies will not be modified.")
```

(further suggestions welcome if you feel this is worse - I feel that directly linking this to the respective option names is useful here)

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased: mitmproxy next
 
 
+Enhance documentation and add alert log messages when stream_large_bodies and modify_body are set (#6479)
```
Comment (mhils) [inline CHANGELOG.md:6]:
```suggestion
* Enhance documentation and add alert log messages when stream_large_bodies and modify_body are set
```

CI will do the linking, but it needs to be a bullet point for that. :)

**Review by mhils — COMMENTED** [review]
Thank you for the super nice first PR! It's great to have both docs updates + an interactive warning. 😃 🍰 🚀 

Super small suggestions below. :)

File: mitmproxy/addons/modifybody.py (hunk 2 above)
Comment (rosydawn6) [reply mitmproxy/addons/modifybody.py]: I have removed this and the corresponding import of ALERT

File: mitmproxy/addons/modifybody.py (hunk 1 above)
Comment (rosydawn6) [reply mitmproxy/addons/modifybody.py:23]: I have reverted this change

File: mitmproxy/addons/modifybody.py (hunk 2 above)
Comment (rosydawn6) [reply mitmproxy/addons/modifybody.py]: Clarifying that I did a git revert

File: CHANGELOG.md (hunk 4 above)
Comment (rosydawn6) [reply CHANGELOG.md:6]: Thanks for that tip.

File: mitmproxy/addons/proxyserver.py (hunk 3 above)
Comment (rosydawn6) [reply mitmproxy/addons/proxyserver.py:216]: I like linking to the option names.   One minor question the message now says that modify_body is active, but we are not checking for its presence.  Are we simply assuming that the body modification could happen via a def response lifecycle event or the modify_body option?

**Comment by rosydawn6** [comment]
Thanks for your suggestions.  I overall agree and have made your changes with one clarifying question.

File: mitmproxy/addons/proxyserver.py (hunk 3 above)
Comment (mhils) [reply mitmproxy/addons/proxyserver.py:216]: Ah, thanks for catching this. We should definitely check if modify_body is set, and only then emit an alert! :)

**Comment by rosydawn6** [comment]
I have added the check for both options.  After some testing I saw that I need to check for both options being set in both addsons as the options can come in any order.

**Review by mhils — APPROVED** [review]
Thanks! 😃 🍰

**Pull request merged**